builder/dockerfile/dispatchers.go

```diff
@@ -346,6 +352,12 @@ func dispatchRun(d dispatchRequest, c *instructions.RunCommand) error {
 	if !system.IsOSSupported(d.state.operatingSystem) {
 		return system.ErrNotSupportedOperatingSystem
 	}
+
+	for _, f := range c.FlagsUsed {
```

Comment:
bit weird to have a return in a for loop. Would be slightly better with an `if len(c.FlagsUsed) > 0 { return errors.Errorf(..., c.FlagsUsed[0]) }`

Reply:
Ah, yes, I think I started with creating an error-message for all flags or to have a "whitelist", but then decided "meh, just return an error for the first one"; I'll update